### ISSUE TITLE
Allow assigning custom THintWindowClass

### ISSUE THREAD

**Issue by IgitBuh** [JAM-Software/Virtual-TreeView#770]
VTV doesn't use the application's HintWindowClass, so if a custom one has been assigned (as easy as `HintWindowClass := TMyHintWindow;`) it's used for all components except for VTV.
Unfortunately, there's no direct way to use a custom hint style, so in order to assign it, one has either to inherit or intercept TVirtualStringTree.

I'm currently doing this:

```
type
  TVirtualStringTree = class(VirtualTrees.TVirtualStringTree)
  public
    function GetHintWindowClass: THintWindowClass; override;
  end;

function TVirtualStringTree.GetHintWindowClass: THintWindowClass;
begin
  Result := TMyHintWindow;
end;
```

It works, but I feel like this shouldn't be necessary.

Would it please be possible to have a public property like "HintWindowClass" to easily assign a custom THintWindowClass?
Alternatively or additionally, there could be an option to automatically use the THintWindowClass of the application, like `toUseApplicationHintWindowClass`.

**Comment by joachimmarder** [comment]
> It works, but I feel like this shouldn't be necessary.

Agreed. I noticed that too a while ago and started digging why VTV needs it own hint window class at all. One outcome was #730.

**Comment by joachimmarder** [comment]
We should remove TVTHintData.DefaultHint and determine TVTHintData.HintText in TBaseVirtualTree.CMHintShow() and not in TVirtualTreeHintWindow.CalcHintRect(). We should aim that the default VCL THintWindow is working, at least header hints are not shown here.

**Comment by joachimmarder** [comment]
The remaining problem in this issue is that the hint is not displayed for the column headers if e.g. the `THintWindow` class is used. The reason for this is that the `WM_NCMOUSEMOVE` message cancels the hint. There is a special handling for this message in `TVirtualTreeHintWindow.IsHintMsg()`.

Easiest way to repro, change:
```
function TVTAncestorVcl.GetHintWindowClass: THintWindowClass;

// Returns the default hint window class used for the tree. Descendants can override it to use their own classes.

begin
  Result := THintWindow;
//  Result :=  TVirtualTreeHintWindow;
end;
```

**Comment by joachimmarder** [comment]
I don't think we can improve anything in the Virtual TreeView code to support displaying of hints in the non-client (NC) area. A developer who wants to use his own hint windows class will need to prepare this by overriding `IsHintMsg()`. I therefore close this issue.
